src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/PlatformWrapperView.java

```diff
@@ -55,17 +158,134 @@ protected void dispatchDraw(Canvas canvas) {
                 if (viewHeight == 0)
                     viewHeight = child.getMeasuredHeight();
             }
-            drawShadow(canvas, viewWidth, viewHeight);
+
+            if (viewWidth > 0 && viewHeight > 0) {
+                drawShadow(canvas, viewWidth, viewHeight);
+            }
         }
+
         super.dispatchDraw(canvas);
     }
 
-    /**
-     * Overridden in C#, for custom logic around shadows
-     * @param canvas
-     * @param viewWidth
-     * @param viewHeight
-     * @return
-     */
-    protected abstract void drawShadow(@NonNull Canvas canvas, int viewWidth, int viewHeight);
+    protected void drawShadow(@NonNull Canvas canvas, int viewWidth, int viewHeight) {
+        if (getChildCount() > 0)
+        {
+            View child = getChildAt(0);
+            Drawable background = child.getBackground();
+            // See if we can quickly draw shadow through Canvas API thanks to the fact we have a solid content
+            if (background != null && background instanceof PlatformShadowDrawable && ((PlatformShadowDrawable)background).canDrawShadow()) {
+                // Layout has already happened on the child view, but not on its drawable, so we need to set the bounds manually
+                int left = child.getLeft();
+                int top = child.getTop();
+                int right = child.getRight();
+                int bottom = child.getBottom();
+                background.setBounds(left, top, right - left, bottom - top);
```

Comment:
```suggestion
                background.setBounds(0, 0, right - left, bottom - top);
```
https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/core/java/android/view/View.java;l=25123;bpv=1;bpt=1